### DISCUSSION TITLE
级联依赖情况下的头文件疑问

### DISCUSSION THREAD

**Discussion by BanKnight** [xmake-io/xmake#2241] [Q&A]
情况依然是改造 txikijs 而来。

## 先介绍下情况，目前有3个xmake.lua
+ libtxikijs工程：这里有多个依赖的target，并且核心target(libtxikijs) 通过  add_packages("libuv") 依赖了libuv
+ repo仓库：对应libtxikijs的包描述 xmake.lua 文件，通过   
``` lua
-- 调用了libtxikijs的xmake
import("package.tools.xmake").install(package, configs)
``` 
+ test_libtxikijs工程：通过 add_packages 依赖了 libtxikijs

## 疑问是

+ 多target情况下，repo如何调用 libtxikijs 这个target
+ libtxikijs 依赖libuv，它提供的头文件需要用到libuv的头文件“uv.h”,如何一并提供出去呢？

**Comment by waruqi** [comment]
libtxikijs 包里，package 定义加上 `add_deps("libuv")`

test_libtxikijs 里 `add_requires("libtxikijs")/add_packages("libtxikijs")`

**Comment by BanKnight** [comment]
测试了一下，的确解决了，赞。

那问题1呢？对于那些提供了多个target的，可能既有binary又有多个libraries的，
我们写对应package时，应该怎么办呢

**Reply by waruqi** [reply]
`import("package.tools.xmake").install(package, configs)` 目前只能全部安装，xmake.lua 里 set_default(false) 的会默认忽略
